src/resource/solidDataset.ts

```diff
@@ -1202,6 +1202,19 @@ export async function getWellKnownSolid(
   > = internal_defaultFetchOptions
 ): Promise<SolidDataset & WithServerResourceInfo> {
   const urlString = internal_toIriString(url);
+
+  // Try to fetch the well-known solid dataset from the pod's root
+  try {
+    const wellKnownSolidUrl = new URL(
+      "/.well-known/solid",
+      new URL(urlString).origin
+    ).href;
+    return await getSolidDataset(wellKnownSolidUrl);
+  } catch (e) {
+    // In case of error, do nothing and try to discover the .well-known
```

Comment:
Would it be worth storing this error in case the rootResource isn't found?

Reply:
I fixed up the commnent of the outer function.
All we know here is the server's root well known solid is not accessible, then we can try the other mechanism, when we're done with it, all we know is neither worked... not which one should have worked... I guess the resource might even not be hosted on a solid server ^^,
So, no, I don't think we need to store the error.

For what it's worth, the behaviour was the same prior (the error was catched and the other mechanism tried out).

Reply:
I think I agree with @matthieubosquet: the error that matters is that none of the `.well-known` discovery mechanism worked as a whole, rather than detailing each failed attempt. This whole thing is implementation-specific anyway, once the discovery mechanism is set at the spec level this will be completely rewritten.